server/src/test/java/org/elasticsearch/common/settings/SettingTests.java

```diff
@@ -659,6 +666,22 @@ public void testCompositeValidator() {
 
     }
 
+    public void testListKeyExists() {
+        final Setting<List<String>> listSetting = Setting.listSetting(
+            "foo",
+            Collections.singletonList("bar"),
+            Function.identity(),
+            Property.NodeScope
+        );
+        Settings settings = Settings.builder().put("foo", "bar1,bar2").build();
+        assertFalse(listSetting.exists(Settings.EMPTY));
+        assertTrue(listSetting.exists(settings));
+
+        settings = Settings.builder().put("foo.0", "foo1").put("foo.1", "foo2").build();
+        assertFalse(listSetting.exists(Settings.EMPTY));
+        assertTrue(listSetting.exists(settings));
```

Comment:
nit: maybe it would be nice (explain usage better) if you assert also on the content (foo1, foo2)?

Reply:
I actually intentionally didn't assert on the content, as it's irrelevant for exists. There's other tests doing that. If you think it's helpful, I can change the value of the content to `ignored` or similar.